packages/fungible-token/bridge-fungible-token/src/cast.sw

```diff
@@ -1,57 +1,50 @@
 library;
 
-// TODO: https://github.com/FuelLabs/sway/issues/4800
-
-use std::u256::U256;
 use std::constants::ZERO_B256;
 
-impl From<b256> for U256 {
+impl From<b256> for u256 {
     fn from(value: b256) -> Self {
-        let (word1, word2, word3, word4) = asm(r1: value) { r1: (u64, u64, u64, u64) };
-        let result = U256::from((word1, word2, word3, word4));
-
-        result
+        value.as_u256()
     }
 
     fn into(self) -> b256 {
-        let result = b256::min();
-
-        asm(output: result, r1: self.a, r2: self.b, r3: self.c, r4: self.d) {
-            sw   output r1 i0; // store the word in r1 in output + 0 words
-            sw   output r2 i1; // store the word in r2 in output + 1 word
-            sw   output r3 i2; // store the word in r3 in output + 1 word
-            sw   output r4 i3; // store the word in r4 in output + 1 word
-        }
-
+        let result: b256 = asm(r1: self) { r1: b256 };
         result
     }
 }
 
-#[test]
-fn test_b256_from() {
-    // Test the boundary conditions: min, middle, max
-    let min = b256::min();
-    let middle = 0x000000000000000000000000000000000000000000000000000000000000000a;
-    let max = b256::max();
-
-    // Alternatively, compare each field of the U256
-    assert_eq(U256::from(min), U256::from((0, 0, 0, 0)));
-    assert_eq(U256::from(middle), U256::from((0, 0, 0, 10)));
-    assert_eq(U256::from(max), U256::from((u64::max(), u64::max(), u64::max(), u64::max())));
-}
+// Weird sway / rust sdk bug does not allow to compile the contract, when enabling this test,
```

Comment:
Should raise an issue, and write in a TODO, prior to merging so that this isn't forgotten

Reply:
Seems to be gone in 0.48, most likely this was a 0.47 issue